### PR TITLE
[ROCM 4.1] load hipfft module for rocm 4.1 and after

### PR DESCRIPTION
**Pull request by micmelesse**
This PR links to hipfft conditionally based on the rcom version.  Based on https://github.com/ROCmSoftwarePlatform/pytorch/pull/775/files

### CODE REVIEW

**Review by jeffdaily — APPROVED** [review]
Let's wait for CI to finish, just in case.

File: cmake/public/LoadHIP.cmake [hunk 1]
```diff
@@ -138,6 +145,26 @@ find_package_and_print_version(HIP 1.0)
...
+    set(ROCM_VERSION_DEV_MAJOR ${CMAKE_MATCH_1})
+    set(ROCM_VERSION_DEV_MINOR ${CMAKE_MATCH_2})
+    set(ROCM_VERSION_DEV_PATCH ${CMAKE_MATCH_3})
+    set(ROCM_VERSION_DEV "${ROCM_VERSION_DEV_MAJOR}.${ROCM_VERSION_DEV_MINOR}.${ROCM_VERSION_DEV_PATCH}")
+  endif()
+  if(ROCM_VERSION_DEV VERSION_GREATER_EQUAL "4.1.0")
+    message("version is >= 4.1.0")
+  else()
+    message("version is < 4.1.0")
+  endif()
```
Comment (jeffdaily) [inline cmake/public/LoadHIP.cmake:161]: we don't need this part any more.  it was just for testing purposes, to make sure the above parsing worked correctly.  the print statements below are good enough to verify parsing worked ok.

Comment (micmelesse) [reply cmake/public/LoadHIP.cmake]: Done

**Pull request merged**